src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1603,12 +1633,22 @@ class Simplex_tree {
       if (dim == 0) return;
       // Find the maximum filtration value in the border
       Boundary_simplex_range&& boundary = boundary_simplex_range(sh);
-      Boundary_simplex_iterator max_border = std::max_element(std::begin(boundary), std::end(boundary),
+      typename SimplexTreeOptions::Filtration_value max_filt_border_value;
+      if constexpr (SimplexTreeOptions::is_multi_parameter){
+        // in that case, we assume that Filtration_value has a `push_to` member to handle this.
```

Comment:
It's a possibility, but I'm only using python, and I will not maintain / deal with a c++ interface that I don't use.

Reply:
It is not about making a C++ interface. Just move the file, include it in `Simplex_tree.h` and add something like:
```C++
struct Simplex_tree_options_multi_parameter {
  typedef linear_indexing_tag Indexing_tag;
  typedef int Vertex_handle;
  typedef Finitely_critical_multi_filtration<float> Filtration_value;
  typedef std::uint32_t Simplex_key;
  static const bool store_key = true;
  static const bool store_filtration = true;
  static const bool contiguous_vertices = false;
  static const bool link_nodes_by_label = false;
  static const bool stable_simplex_handles = false;
  static const bool is_multi_parameter = true;
};
```
at the end of the file with the other options instead of having it in `/python/include/Simplex_tree_multi.h`. That is the only change. Then for the documentation, in the concept of the options (concept/SimplexTreeOptions.h) you mention the need of `push_to` and that you can find an example of implementation as ` Finitely_critical_multi_filtration<T>`.

The difference is that it will make the tree usable for C++ users without you having to change much or to maintain anything.

Reply:
@mglisse Not really, it was just to illustrate how we would use it. If Filtration_value needs a special method like `push_to`, we cannot just let it hang there without example for the user, in particular if code was already written. We can also put this just in the tests or in an example, but at least in the concept, the implementation of `Finitely_critical_multi_filtration<T>` should be mentioned.